docs/copilot/mcp-servers.md

```diff
@@ -0,0 +1,211 @@
+            }
+        },
+        // https://github.com/modelcontextprotocol/servers/tree/main/src/fetch
+        "fetch": {
+            "type": "stdio",
+            "command": "uvx",
+            "args": ["mcp-server-fetch"]
+        },
+        "my-remote-server": {
+            "type": "sse",
+            "url": "http://api.contoso.com/sse",
+            "headers": { "VERSION": "1.2" }
+        }
+    }
+}
+```
+
+## Use MCP tools in agent mode
+
+Once you have added an MCP server, you can use the tools it provides in agent mode. To use MCP tools in agent mode:
+
+1. Make sure you're using [VS Code Insiders](http://code.visualstudio.com/insiders).
+
+1. Open the **Chat** view (`kb(workbench.action.chat.open)`), and select **Agent** mode from the dropdown.
+
+    ![Agent mode dropdown option](images/copilot/mcp-tool-list.png)
+
+1. Select the **Refresh** button to start and refresh the list of available MCP servers.
+
+    ![Refresh button](images/copilot/mcp-tool-list.png)
+
+1. Select the tools button to view the list of available tools.
+
+    Optionally, select or deselect the tools you want to use.
+
+    ![MCP tools list](images/copilot/mcp-tool-list.png)
+
+1. You can now enter a prompt in the chat input box and notice how the tools are automatically invoked as needed.
+
+    By default, when a tool is invoked, you need to confirm the action before it is run. This is because tools might run locally on your machine and might perform actions that modify files or data.
+
+    Use the **Continue** button dropdown options to automatically confirm the specific tool for the current session, workspace, or all future invocations.
+
+    ![MCP Tool Confirmation](images/copilot/mcp-tool-confirmation.png)
+
+## Managing tools
+
+Run the **MCP: List Servers** command from the Command Palette to view the list of configured MCP servers.
+
+![MCP Server List](images/copilot/mcp-server-list.png)
+
+When you select a server, you can start, stop, or restart the server. You can also view the server configuration and server logs to diagnose issues.
+
+> [!TIP]
+> When you open the `.vscode/mcp.json` file, VS Code shows commands to start, stop, or restart a server directly from the editor.
+
+![MCP server configuration with lenses to manage server.](images/copilot/mcp-server-config.png)
+
+### Command-line configuration
+
+You can also use the VS Code command-line interface to add an MCP server to your user profile or to a workspace.
+
+To add an MCP server to your user profile, use the `--add-mcp` command line option, and provide the JSON server configuration in the form `{\"name\":\"server-name\",\"command\":...}`.
+
+```bash
+code --add-mcp "{\"name\":\"my-server\",\"command\": \"uvx\",\"args\": [\"mcp-server-fetch\"]}"
+```
+
+To add an MCP server to a workspace, in addition to the `--add-mcp` option, also specify the `--mcp-workspace` option and provide the folder or workspace in which to add the server.
```

Comment:
`--mcp-workspace` is not there yet, we need some additional features from the configuration service